ghcide/src/Development/IDE/Core/Tracing.hs

```diff
@@ -77,21 +78,27 @@ otTracedAction
     :: Show k
     => k -- ^ The Action's Key
     -> NormalizedFilePath -- ^ Path to the file the action was run for
-    -> (a -> Bool) -- ^ Did this action succeed?
-    -> Action a -- ^ The action
-    -> Action a
-otTracedAction key file success act
+    -> RunMode
+    -> (a -> Bool)
+    -> Action (RunResult a) -- ^ The action
+    -> Action (RunResult a)
+otTracedAction key file mode success act
   | userTracingEnabled =
     actionBracket
         (do
             sp <- beginSpan (fromString (show key))
             setTag sp "File" (fromString $ fromNormalizedFilePath file)
+            setTag sp "Mode" (fromString $ show mode)
             return sp
         )
         endSpan
         (\sp -> do
             res <- act
-            unless (success res) $ setTag sp "error" "1"
+            unless (success $ runValue res) $ setTag sp "error" "1"
+            setTag sp "changed" $ case res of
+              RunResult ChangedRecomputeSame _ _ -> "0"
+              RunResult ChangedNothing _ _       -> "0"
+              RunResult ChangedRecomputeDiff _ _ -> "0"
```

Comment:
This pattern match is non-exhaustive, though we never use `ChangedStore`. BTW why the result of these three cases is constant `"0"`?

Reply:
That was a mistake during cleanup, corrected now. Thanks!